### PR TITLE
ref(dam): Use cols and aggregates

### PR DESCRIPTION
**Pull request by shruthilayaj**
Read from columns and aggregates instead of fields to
build widgets. Fields are still used to maintain order on
table widgets.

### CODE REVIEW

**Review by matejminar — APPROVED** [review]
Looks good to me.

I feel like some components accept the `fields` prop which we build by combining `columns` and `aggregates` only to later down the road try to get from `fields` back to `columns` and `aggregates`. Maybe something we should address in the future.

File: tests/sentry/api/endpoints/test_organization_dashboard_details.py [hunk 1]
```diff
@@ -36,20 +36,26 @@ def setUp(self):
...
+            columns=self.anon_users_query["columns"],
+            aggregates=self.anon_users_query["aggregates"],
             conditions=self.anon_users_query["conditions"],
             order=0,
         )
         self.widget_1_data_2 = DashboardWidgetQuery.objects.create(
             widget=self.widget_1,
             name=self.known_users_query["name"],
             fields=self.known_users_query["fields"],
+            columns=self.anon_users_query["columns"],
```
Comment (priscilawebdev) [inline tests/sentry/api/endpoints/test_organization_dashboard_details.py:48]: should we have backend changes in a separate PR?

File: static/app/components/modals/addDashboardWidgetModal.tsx [hunk 2]
```diff
@@ -428,12 +434,13 @@ class AddDashboardWidgetModal extends React.Component<Props, State> {
...
             // When switching back to original display type, default fields back to the fields provided from the discover query
             normalized.forEach(query => {
-              query.fields = [...defaultWidgetQuery.fields];
-              const {columns, aggregates} = getColumnsAndAggregates([
-                ...defaultWidgetQuery.fields,
-              ]);
-              query.aggregates = aggregates;
-              query.columns = columns;
+              query.aggregates = [...defaultWidgetQuery.aggregates];
+              query.columns = [...defaultWidgetQuery.columns];
```
Comment (priscilawebdev) [inline static/app/components/modals/addDashboardWidgetModal.tsx:438]: please can we add these changes to the `widgetBuilder` too?

File: tests/sentry/api/endpoints/test_organization_dashboard_details.py (hunk 1 above)
Comment (shruthilayaj) [reply tests/sentry/api/endpoints/test_organization_dashboard_details.py:48]: yup, I have a backend PR open, can you take a look? https://github.com/getsentry/sentry/pull/32518 @priscilawebdev

File: static/app/views/dashboardsV2/widgetBuilder/columnFields.tsx [hunk 3]
```diff
@@ -39,17 +43,17 @@ export function ColumnFields({
...
           organization={organization}
           source={widgetType}
         />
       ) : (
         <ColumnCollectionEdit
-          columns={columns.slice(0, columns.length - 1)}
+          columns={[...columns, ...aggregates.slice(0, aggregates.length - 1)]}
           onChange={newColumns => {
-            onChange([...newColumns, columns[columns.length - 1]]);
+            onChange([...newColumns, fields[fields.length - 1]]);
```
Comment (narsaynorath) [inline static/app/views/dashboardsV2/widgetBuilder/columnFields.tsx:56]: Can I ask, what's the significance of `[fields.length - 1]` here?

This is when we're not in a table, and I think this component shows up in Top-N, so what I'm gathering is that the last index in fields is our Y-Axis for a Top-N widget and everything else shows up in the Columns section

File: static/app/views/dashboardsV2/widgetBuilder/widgetBuilder.tsx [hunk 4]
```diff
@@ -461,7 +459,41 @@ function WidgetBuilder({
...
+        } else {
+          newQuery.orderby = '';
+        }
+      }
+
+      handleQueryChange(queryIndex, newQuery);
+    }
+  }
+
   function handleYAxisOrColumnFieldChange(newFields: QueryFieldValue[]) {
```
Comment (narsaynorath) [inline static/app/views/dashboardsV2/widgetBuilder/widgetBuilder.tsx:495]: If we've created our own `handleYAxisChange`, should this become `handleColumnFieldChange`?

**Review by narsaynorath — APPROVED** [review]
Just some small comments but the changes look good to me. Thanks for taking this on and separating the columns and aggregates!

File: static/app/views/dashboardsV2/widgetBuilder/columnFields.tsx (hunk 3 above)
Comment (shruthilayaj) [reply static/app/views/dashboardsV2/widgetBuilder/columnFields.tsx:56]: yep, you're right, it's how we get the y axis for top n

**Pull request merged**